superset/assets/src/explore/components/controls/TimeSeriesColumnControl.jsx

```diff
@@ -52,7 +53,21 @@ const colTypeOptions = [
 export default class TimeSeriesColumnControl extends React.Component {
   constructor(props) {
     super(props);
-    const state = { ...props };
+    const state = {
+      label: this.props.label,
+      tooltip: '',
+      colType: '',
+      width: '',
+      height: '',
+      timeLag: '',
+      timeRatio: '',
+      comparisonType: '',
+      showYAxis: false,
+      yAxisBounds: [null, null],
+      bounds: [null, null],
+      d3format: '',
+      dateFormat: '',
+    };
```

Comment:
Previously state was set to what's in props `const state = { ...props }`, should these be set to the values in props instead of nulls and empty strings? It would also be good to explicitly state what's in props and list them as either required or list them in defaultProps.

Reply:
None of the properties in props correspond to state properties except label.

<img width="510" alt="Screen Shot 2019-04-15 at 12 56 55 PM" src="https://user-images.githubusercontent.com/47833996/56161311-0b343680-5f7e-11e9-907d-968e0cd8b0bb.png">

The line I removed is definitely confusing to read. It leads the reader to believe that props has values like "colType", "width", etc. when in actuality it has none of these properties.

Listing out the state properties should improve readability.

I've also added a default value for label in props.

Reply:
You can also see in the screenshot all the random properties we add to state that leads to a bad request to the backend.

Reply:
Instead of adding `null` values and empty strings, can we just _omit_ them? The consumer of that object should be the thing that decides a) if it wants those values to be there and b) what to do if they are missing. Inserting them here seems like it is coupling things in a way that is going to be fragile

Reply:
Nvm, @khtruong I think that works just as a declaration of "here are the fields and initial values that this type cares about".  I was thinking it was being used differently.

Reply:
Are you printing out the this.props from `CollectionControl`? When I look at props from within the constructor of `TimeSeriesColumnControl` when loading a time series table chart I see the other props like `colType` `d3format` set that correspond to state properties. You mentioned in chat that `o` passed to the `Control` is undefined, on master it looks like it's getting set correctly. Are there any changes in lyftga that could have affected this?

Reply:
Ahh I see what is going on. I was looking into this.props and you were referring to this.props.value. Yes, I can set the state to what is stored in this.props.values. 

Also ignore my comment about o being undefined. I was within the map function's scope when I tried to look up this.props which was undefined within that scope.

Reply:
Talked to Michelle offline. Turns out she was looking at the props with Max's changes and I was looking at the props without Max's changes. Hehe.

Now I have set state to the prop values if they exist. Otherwise, they are sent to default values.